compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3077,6 +3077,15 @@ private void checkObjectFunctionInvocationExpr(BLangInvocation iExpr, BObjectTyp
                 return;
             }
         }
+
+        // __init method can be called in a method-call-expr only when the expression
+        // preceding the . is self.
+        if (iExpr.name.value.equals(Names.USER_DEFINED_INIT_SUFFIX.value) &&
+                !(iExpr.expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF &&
+                ((BLangSimpleVarRef) iExpr.expr).getVariableName().value.equals(Names.SELF.value))) {
```

Comment:
We usually use the symbol for these checks. e.g.,
```java
Names.SELF.equals(((BLangSimpleVarRef) iExpr.expr).symbol.name)
```

Reply:
Changed.